waffle_hub/hub/hub.py

```diff
@@ -884,21 +908,51 @@ def inner(callback: TrainCallback, result: TrainResult):
             device=device,
             workers=workers,
             seed=seed,
+            advance_params=advance_params if advance_params else {},
             verbose=verbose,
         )
 
-        # overwrite train config with default config
+        ## overwrite train config with default config
         for k, v in cfg.to_dict().items():
             if v is None:
                 field_value = getattr(
                     self.DEFAULT_PARAMS[self.task][self.model_type][self.model_size], k
                 )
                 setattr(cfg, k, field_value)
 
+        ## overwrite train advance config
+        if cfg.advance_params:
+            if isinstance(cfg.advance_params, (str, PurePath)):
```

Comment:
As far as I know, PurePath inherits from Path, but is there a specific reason why it must be of type PurePath?

Reply:
`PurePath` is base class of `Path`s of `pathlib`.